examples/threading/threading.c

```diff
@@ -14,6 +18,53 @@ void* threadfunc(void* thread_param)
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    int ret,ret1;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param; //obtaining thread arguments through parameter
+    
+    if((thread_func_args->wait_to_obtain_ms)<0)
```

Comment:
@chth2844  you could make `wait_to_obtain_ms` an unsigned int to avoid the need for this check